java-desktop/src/main/java/com/couchbase/todo/controller/ConfigController.java

```diff
@@ -0,0 +1,106 @@
+package com.couchbase.todo.controller;
+
+import java.net.URL;
+import java.util.ResourceBundle;
+
+import javafx.fxml.FXML;
+import javafx.fxml.Initializable;
+import javafx.scene.control.Button;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.TextField;
+import javafx.stage.Stage;
+import org.jetbrains.annotations.NotNull;
+
+import com.couchbase.lite.internal.core.CBLVersion;
+import com.couchbase.todo.TodoApp;
+import com.couchbase.todo.model.Config;
+import com.couchbase.todo.model.DB;
+
+
+public class ConfigController implements Initializable {
+    private @NotNull final Stage stage;
+
+    @FXML
+    private CheckBox loggingCheckbox;
+    @FXML
+    private CheckBox loginCheckbox;
+    @FXML
+    private CheckBox localCheckbox;
+    @FXML
+    private CheckBox remoteCheckbox;
+    @FXML
+    private TextField cblVersionTextField;
+    @FXML
+    private TextField dbNameTextField;
+    @FXML
+    private TextField sgUrlTextField;
+    @FXML
+    private TextField maxRetriesTextField;
+    @FXML
+    private TextField waitTimeTextField;
+    @FXML
+    private Button saveButton;
+    @FXML
+    private Button cancelButton;
+
+
+    public ConfigController(@NotNull Stage stage) { this.stage = stage; }
+
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+        //set initial values of checkbox and text field
+        RegisterButtonEventHandler();
+
+        final Config config = TodoApp.getTodoApp().getConfig();
+        cblVersionTextField.setText(CBLVersion.getVersionInfo());
+
+        loggingCheckbox.setSelected(config.isLoggingEnabled());
+        loginCheckbox.setSelected(config.isLoginRequired());
+
+        setCcrState(config.getCrMode());
+
+        dbNameTextField.setText(config.getDbName());
+        sgUrlTextField.setText(config.getSgwUri());
+        maxRetriesTextField.setText(String.valueOf(config.getAttempts()));
+        waitTimeTextField.setText(String.valueOf(config.getAttemptsWaitTime()));
+    }
+
+    private void RegisterButtonEventHandler() {
+        // handle events clicking cancel button and save buttons
+        cancelButton.setOnAction(event -> TodoApp.goToPage(this.stage, TodoApp.MAIN_FXML));
+
+        saveButton.setOnAction(event -> {
+            update();
+            TodoApp.goToPage(this.stage, TodoApp.LOGIN_FXML);
+        });
+    }
+
+    private void update() {
+        String eDbName = (dbNameTextField.getText().isEmpty()) ? null : dbNameTextField.getText();
```

Comment:
FWIW, I don't see a need for these variables.  If it were me, writing this, I'd just say:
Config.builder()
     // ...
    .dbName((dbNameTextField.getText().isEmpty()) ? null : dbNameTextField.getText())